src/main/java/com/onfleet/demo/homework/util/ObjectGenerator.java

```diff
@@ -127,7 +130,33 @@ public double getLongitudeHiMark() {
    * @return Random Location.
    */
   static @NotNull Geopoint randomLocationWithinBounds() {
-    return new Geopoint(0.0, 0.0);  // @TODO: real code
+    // quick and dirty algorithm to generate roughly within the points
+    final Random randomGenerator = RandomNumberUtil.getRandomGenerator();
+    final double morePreciseLowLatitude = geopointWatermarks.getLatitudeLowMark() * Watermarks.precisionMultiple;
+    final double morePreciseLowLongitude = geopointWatermarks.getLongitudeLowMark() * Watermarks.precisionMultiple;
+    final double morePreciseHiLatitude = geopointWatermarks.getLatitudeHiMark() * Watermarks.precisionMultiple;
+    final double morePreciseHiLongitude = geopointWatermarks.getLongitudeHiMark() * Watermarks.precisionMultiple;
+
+    double xValue = Math.abs(morePreciseLowLatitude) + randomGenerator.nextInt(Math.abs((int)morePreciseHiLatitude / 100));
+    double yValue = Math.abs(morePreciseLowLongitude) + randomGenerator.nextInt(Math.abs((int)morePreciseHiLongitude / 100));
+
+    // rein in above-max values by reducing them by (up to 80% of the total spillage over-max plus the spillage)
+    if (Math.abs(xValue) > Math.abs(morePreciseHiLatitude))
+      xValue -= ((Math.abs(xValue) - Math.abs(morePreciseHiLatitude))
+                  + (((double)randomGenerator.nextInt(80)) / 100.0) * (Math.abs(xValue) - Math.abs(morePreciseHiLatitude)));
+    if (Math.abs(yValue) > Math.abs(morePreciseHiLongitude))
+      yValue -= (Math.abs(yValue) - Math.abs(morePreciseHiLongitude))
+                  + (((double)randomGenerator.nextInt(80)) / 100.0) * (Math.abs(yValue) - Math.abs(morePreciseHiLongitude));
+
+    xValue /= Watermarks.precisionMultiple;
+    yValue /= Watermarks.precisionMultiple;
+
+    if (morePreciseLowLatitude < 0.0)
+      xValue = 0.0 - xValue;
+    if (morePreciseLowLongitude < 0.0)
+      yValue = 0.0 - yValue;
+
+    return new Geopoint(xValue, yValue);
```

Comment:
quick and dirty algorithm to generate geopoints, could easily be optimized further to avoid some math